06_puppy/kasunfdo/types.go

```diff
@@ -0,0 +1,29 @@
+package main
+
+import (
+	"sync"
+)
+
+type Puppy struct {
+	ID     uint64
+	Breed  string
+	Colour string
+	Value  float64
+}
+
+type Storer interface {
+	CreatePuppy(puppy Puppy) uint64
+	ReadPuppy(id uint64) (Puppy, bool)
+	UpdatePuppy(id uint64, puppy Puppy) bool
+	DeletePuppy(id uint64) bool
+}
+
+type MapStore struct {
+	store map[uint64]Puppy
+	count uint64
+}
+
+type SyncStore struct {
+	store sync.Map
+	count uint64
+}
```

Comment:
IMO SyncStore & MapStore should sit with their implementation files (`syncstore.go`, `mapstore.go`).
Also, consider renaming `count` to `nextID` (that's how it seems to be used).

Reply:
Agree. :)

Reply:
Agree